### PR TITLE
Added additional colour options for differing git staging statuses

### LINKED ISSUE #99129: Theme Color : Git : Staged
For better visibility it would be nice to add different color settings for "staged" files.
Like: 
- "gitDecoration.stagedModifiedResourceForeground"
- "gitDecoration.stagedAddedResourceForeground"

### PR DESCRIPTION
**Pull request by mwood77**
This PR fixes #99129 - added the ability to colourize different git staging statuses.

The staged statuses include:
- Add (index added)
- Modified (index modified)
- Deleted (index deleted)

The default colours have been set to be similar, albeit slightly different tha what the conventional `U, A, M, D` git indicator colours. This way users can, at a glance, see the different statuses.

Screenshot 2020-09-23 at 20 14 57

### CODE REVIEW

File: extensions/git/package.json [hunk 1]
```diff
@@ -1922,6 +1922,33 @@
           "highContrast": "#E2C08D"
         }
       },
+      {
+        "id": "gitDecoration.stageAddedResourceForeground",
```
Comment (joaomoreno) [inline extensions/git/package.json:1926]: `stageAdded` doesn't exist. There's `untracked` and there's `added`, both states already have individual colors.

File: extensions/git/package.nls.json [hunk 2]
```diff
@@ -168,6 +168,9 @@
 	"submenu.stash": "Stash",
 	"colors.added": "Color for added resources.",
 	"colors.modified": "Color for modified resources.",
+	"colors.stageAdded": "Color for resources which have been staged.",
+	"colors.stageModified": "Color for resources which have been staged, but modified.",
+	"colors.stageDeleted": "Color for resources which have been staged, but deleted.",
```
Comment (joaomoreno) [inline extensions/git/package.nls.json:173]: There's got to be a better description here.

File: extensions/git/package.json [hunk 3]
```diff
@@ -1922,6 +1922,33 @@
...
+          "highContrast": "#ca541d"
+        }
+      },
+      {
+        "id": "gitDecoration.stageDeletedResourceForeground",
+        "description": "%colors.stageDeleted%",
+        "defaults": {
+          "light": "#ad0707",
+          "dark": "#c74e39",
+          "highContrast": "#c74e39"
```
Comment (joaomoreno) [inline extensions/git/package.json:1949]: I don't think we should change the default colors, so let's just use today's colors. This is about enabling themes to overwrite this, not to change the default themes.

**Review by joaomoreno — APPROVED** [review]
I've addressed my comments with a commit on top and will merge it.

**Comment by joaomoreno** [comment]
Thanks! 🍻

**Comment by mwood77** [comment]
Thanks for the feedback!

**Comment by bpasero** [comment]
@joaomoreno @mwood77 please make sure to document this new color here: https://github.com/microsoft/vscode-docs/blob/73633a2eb1e302903f29523f88265c755417a5b0/api/references/theme-color.md#L684-L684

**Comment by mwood77** [comment]
raised @ https://github.com/microsoft/vscode-docs/pull/4047

**Comment by joaomoreno** [comment]
@bpasero I already did this 3 days ago on `vnext`: https://github.com/microsoft/vscode-docs/commit/6b9a0ac47c37b9431d58ccd328961eb80690b07f

**Comment by bpasero** [comment]
Thanks!

**Pull request merged**